assets/js/modules/idea-hub/datastore/idea-state.js

```diff
@@ -78,36 +78,38 @@ const fetchPostUpdateIdeaStateStore = createFetchStore( {
 	reducerCallback: ( state, idea ) => {
 		const { newIdeas = [], savedIdeas = [] } = state;
 
+		const optIn = ( { _id } ) => _id === idea._id;
+		const optOut = ( { _id } ) => _id !== idea._id;
+
 		if ( idea.dismissed === true ) {
 			return {
 				...state,
-				newIdeas: newIdeas.filter( ( { name } ) => name !== idea.name ),
+				newIdeas: newIdeas.filter( optOut ),
+				savedIdeas: savedIdeas.filter( optOut ),
 			};
-		}
-
-		if ( idea.saved === true ) {
-			const ideaDetails = newIdeas.filter(
-				( { name } ) => name === idea.name
-			);
-			if ( ! ideaDetails.length ) {
-				return state;
+		} else if ( idea.saved === true ) {
+			let ideaDetails = newIdeas.find( optIn );
+			if ( ! ideaDetails ) {
+				ideaDetails = idea;
 			}
 
 			return {
 				...state,
-				newIdeas: newIdeas.filter( ( { name } ) => name !== idea.name ),
-				savedIdeas: [ ...savedIdeas, ...ideaDetails ],
-			};
-		} else if ( idea.saved === false ) {
```

Comment:
Don't we still need a condition here for `idea.saved === false`? When you unsave an idea in the "Saved" tab, it should disappear from there and then new ideas should be updated so that it reappears there _if_ it still is a new idea.

Reply:
Yes, it worked like that. Added back `else if ( idea.saved === false )` condition to be more compatible with how it's done in the develop branch.

Reply:
Hmm, it looks like for some reason the response doesn't include the property if it is set to `false`. Try unsaving an idea from the saved tab - even though it sends `saved:false` seemingly correctly, the response only includes `name` and `_id`, and therefore the logic here isn't triggered. Unsaving itself actually works as expected though, on the following page load you see the idea in the right place.

Could it be that somehow the `saved: false` is stripped on the way to our data point logic, since it's `false`? If not, this may be an API bug we need to flag, unless you see another reason it's occuring. While we could work around it, I don't think that's a good idea, it should rather be fixed in the API then.

Reply:
> Could it be that somehow the saved: false is stripped on the way to our data point logic, since it's false?
No, I don't think so. I have created a follow-up PR (#3818) that fixes this issue on our end.